### PR TITLE
Decorators

### PR DESCRIPTION
**Pull request by nicholasruunu**

### CODE REVIEW

File: src/Hydrator/ModelHydrator.php [hunk 1]
```diff
@@ -13,30 +13,23 @@
  */
 final class ModelHydrator implements Hydrator
 {
+    private $arrayHydrator;
+    private $creatableFromArray;
+
+    public function __construct(ArrayHydrator $arrayHydrator, CreatableFromArray $creatableFromArray)
```
Comment (Nyholm) [inline src/Hydrator/ModelHydrator.php:19]: Createable from array is a Model. 
Do you propose one Hydrator instance per model?

File: src/Hydrator/ModelHydrator.php [hunk 2]
```diff
@@ -13,30 +13,23 @@
...
-        }
-
-        if (is_subclass_of($class, CreatableFromArray::class)) {
-            $object = call_user_func($class.'::createFromArray', $data);
-        } else {
-            $object = new $class($data);
-        }
-
-        return $object;
+        return $this->creatableFromArray->createFromArray(
```
Comment (Nyholm) [inline src/Hydrator/ModelHydrator.php:31]: This assume that the model has been instanciated.

**Review by Nyholm — COMMENTED** [review]
Can you put this PR in context. How are you supposed to use the ModelHydrator?

File: src/Hydrator/ModelHydrator.php (hunk 1 above)
Comment (nicholasruunu) [reply src/Hydrator/ModelHydrator.php:19]: What's the difference with passing $class? Don't you need other implementations of CreatableFromArray anyway?

Comment (Nyholm) [reply src/Hydrator/ModelHydrator.php:19]: Of course you do. Check my PR. Where is the hydrator instanciated? Which classes implement CreatableFromArray? 

With this, you cannot reuse the same hydrator for two endpoints.

**Pull request closed without merging**